frontend/src/container/NewWidget/RightContainer/Threshold/Threshold.tsx

```diff
@@ -0,0 +1,262 @@
+	const [color, setColor] = useState<string>(thresholdColor);
+	const [format, setFormat] = useState<ThresholdProps['thresholdFormat']>(
+		thresholdFormat,
+	);
+
+	const isDarkMode = useIsDarkMode();
+
+	const saveHandler = (): void => {
+		setIsEditMode(false);
+		if (setThresholds === undefined) {
+			return;
+		}
+		setThresholds((prevThresholds) =>
+			prevThresholds.map((threshold) => {
+				if (threshold.index === index) {
+					return {
+						...threshold,
+						isEditEnabled: false,
+						thresholdColor: color,
+						thresholdFormat: format,
+						thresholdOperator: operator as ThresholdProps['thresholdOperator'],
+						thresholdUnit: unit,
+						thresholdValue: value,
+					};
+				}
+				return threshold;
+			}),
+		);
+	};
+
+	const editHandler = (): void => {
+		setIsEditMode(true);
+	};
+
+	const handleOperatorChange = (value: string | number): void => {
+		setOperator(value);
+	};
+
+	const handleValueChange = (value: number | null): void => {
+		if (value === null) {
+			return;
+		}
+		setValue(value);
+	};
+
+	const handleUnitChange = (value: string): void => {
+		setUnit(value);
+	};
+
+	const handlerFormatChange = (
+		value: ThresholdProps['thresholdFormat'],
+	): void => {
+		setFormat(value);
+	};
+
+	const deleteHandler = (): void => {
+		if (thresholdDeleteHandler) {
+			thresholdDeleteHandler(index);
+		}
+	};
+
+	const ref = useRef<HTMLDivElement>(null);
+	const [{ handlerId }, drop] = useDrop<
+		ThresholdProps,
+		void,
+		// eslint-disable-next-line @typescript-eslint/no-explicit-any
+		{ handlerId: any }
+	>({
+		accept: 'Threshold',
+		collect(monitor) {
+			return {
+				handlerId: monitor.getHandlerId(),
+			};
+		},
+		hover(item: ThresholdProps, monitor) {
+			if (!ref.current) {
+				return;
+			}
+			const dragIndex = item.keyIndex;
+			const hoverIndex = keyIndex;
+
+			if (dragIndex === hoverIndex) {
+				return;
+			}
+
+			const hoverBoundingRect = ref.current?.getBoundingClientRect();
+
+			const hoverMiddleY = (hoverBoundingRect.bottom - hoverBoundingRect.top) / 2;
+
+			const clientOffset = monitor.getClientOffset();
+
+			const hoverClientY = (clientOffset as XYCoord).y - hoverBoundingRect.top;
+
+			if (dragIndex < hoverIndex && hoverClientY < hoverMiddleY) {
+				return;
+			}
+
+			if (dragIndex > hoverIndex && hoverClientY > hoverMiddleY) {
+				return;
+			}
+
+			moveThreshold(dragIndex, hoverIndex);
+			// eslint-disable-next-line no-param-reassign
+			item.keyIndex = hoverIndex;
+		},
+	});
+
+	const [{ isDragging }, drag] = useDrag({
+		type: 'Threshold',
+		item: () => ({ keyIndex }),
+		collect: (monitor) => ({
+			isDragging: monitor.isDragging(),
+		}),
+	});
+
+	const opacity = isDragging ? 0 : 1;
+	drag(drop(ref));
+
+	return (
+		<div
+			ref={ref}
+			style={{ opacity }}
+			data-handler-id={handlerId}
+			className="threahold-container"
+		>
+			<Card
+				className={
+					isDarkMode
+						? `threahold-card threahold-card-dark`
```

Comment:
rename all instances of `threahold*` to `threshold`

Reply:
Done.